### DISCUSSION TITLE
how to get more info no this error?

### DISCUSSION THREAD

**Discussion by sibelius** [Automattic/mongoose#14267] [General]
how to get more info no this error?

the document value looks correct, we don't have a `path` `timestamp`

```
CastError: Cast to date failed for value "NaN" (type number) at path "timestamp"
at SchemaDate.cast (/app/node_modules/mongoose/lib/schema/date.js:376:11)
at SchemaType.applySetters (/app/node_modules/mongoose/lib/schemaType.js:1219:12)
at SchemaDate.castForQuery (/app/node_modules/mongoose/lib/schema/date.js:410:17)
at castUpdateVal (/app/node_modules/mongoose/lib/helpers/query/castUpdate.js:569:19)
at walkUpdatePath (/app/node_modules/mongoose/lib/helpers/query/castUpdate.js:374:24)
at castUpdate (/app/node_modules/mongoose/lib/helpers/query/castUpdate.js:107:7)
at model.Query._castUpdate (/app/node_modules/mongoose/lib/query.js:4569:10)
at model.Query._findOneAndUpdate (/app/node_modules/mongoose/lib/query.js:3260:23)
at model.Query.exec (/app/node_modules/mongoose/lib/query.js:4290:80)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  stringValue: '"NaN"',
    messageFormat: undefined,
    kind: 'date',
    value: NaN,
    path: 'timestamp',
    reason: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

    assert.ok(false)

  at castDate (/app/node_modules/mongoose/lib/cast/date.js:40:10)
  at SchemaDate.cast (/app/node_modules/mongoose/lib/schema/date.js:374:12)
  at SchemaType.applySetters (/app/node_modules/mongoose/lib/schemaType.js:1219:12)
  at SchemaDate.castForQuery (/app/node_modules/mongoose/lib/schema/date.js:410:17)
  at castUpdateVal (/app/node_modules/mongoose/lib/helpers/query/castUpdate.js:569:19)
  at walkUpdatePath (/app/node_modules/mongoose/lib/helpers/query/castUpdate.js:374:24)
  at castUpdate (/app/node_modules/mongoose/lib/helpers/query/castUpdate.js:107:7)
  at model.Query._castUpdate (/app/node_modules/mongoose/lib/query.js:4569:10)
  at model.Query._findOneAndUpdate (/app/node_modules/mongoose/lib/query.js:3260:23)
  at model.Query.exec (/app/node_modules/mongoose/lib/query.js:4290:80) {
    generatedMessage: true,
      code: 'ERR_ASSERTION',
      actual: false,
      expected: true,
      operator: '=='
  },
  valueType: 'number'
}

```

can we add more info? document data, stacktrace from the place that happened this cast error

**Comment by sibelius** [comment]
wrong date in a date field

**Reply by vkarpov15** [reply]
The issue is that the stack trace has exactly 10 elements, which is Node's default stack trace limit. For example, the following script will trigger the exact same error:

```javascript
'use strict';

const mongoose = require('mongoose');

run().catch(err => console.log(err));

async function run() {
  await mongoose.connect('mongodb://127.0.0.1:27017/mongoose_test');
  const schema = new mongoose.Schema({ test: Date });
  const TestModel = mongoose.model('Test', schema);

  await TestModel.findOneAndUpdate({}, { test: NaN });
}
```

By default you get a stack trace with only 10 lines:

```
CastError: Cast to date failed for value "NaN" (type number) at path "test"
    at SchemaDate.cast (/mongoose/lib/schema/date.js:376:11)
    at SchemaType.applySetters (/mongoose/lib/schemaType.js:1221:12)
    at SchemaDate.castForQuery (/mongoose/lib/schema/date.js:411:19)
    at castUpdateVal (/mongoose/lib/helpers/query/castUpdate.js:573:19)
    at walkUpdatePath (/mongoose/lib/helpers/query/castUpdate.js:378:24)
    at castUpdate (/mongoose/lib/helpers/query/castUpdate.js:111:7)
    at model.Query._castUpdate (/mongoose/lib/query.js:4548:10)
    at model.Query._findOneAndUpdate (/mongoose/lib/query.js:3261:23)
    at model.Query.exec (/mongoose/lib/query.js:4291:80)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

But add `Error.stackTraceLimit = 15` to the top of the js file and you get the following, which includes the line in my repro script that originally called `findOneAndUpdate()`.

```
CastError: Cast to date failed for value "NaN" (type number) at path "test"
    at SchemaDate.cast (/mongoose/lib/schema/date.js:376:11)
    at SchemaType.applySetters (/mongoose/lib/schemaType.js:1221:12)
    at SchemaDate.castForQuery (/mongoose/lib/schema/date.js:411:19)
    at castUpdateVal (/mongoose/lib/helpers/query/castUpdate.js:573:19)
    at walkUpdatePath (/mongoose/lib/helpers/query/castUpdate.js:378:24)
    at castUpdate (/mongoose/lib/helpers/query/castUpdate.js:111:7)
    at model.Query._castUpdate (/mongoose/lib/query.js:4548:10)
    at model.Query._findOneAndUpdate (/mongoose/lib/query.js:3261:23)
    at model.Query.exec (/mongoose/lib/query.js:4291:80)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async run (/troubleshoot-mongoose/gh-14270.js:14:3)
```

So TLDR: crank up `Error.stackTraceLimit` because unfortunately Mongoose's stack trace is a little too deep for Node.js' default in this case.
